src/components/calculator/calculator.component.jsx

```diff
@@ -1,56 +1,76 @@
         super(props);
-    
-        this.state = 
-              [
-                { text: 'AC', type: 'normal' },
-                { text: '+/-', type: 'normal' },
-                { text: '%', type: 'normal' },
-                { text: '÷', type: 'orange' },
-                { text: '7', type: 'normal' },
-                { text: '8', type: 'normal' },
-                { text: '9', type: 'normal' },
-                { text: 'x', type: 'orange' },
-                { text: '4', type: 'normal' },
-                { text: '5', type: 'normal' },
-                { text: '6', type: 'normal' },
-                { text: '-', type: 'orange' },
-                { text: '1', type: 'normal' },
-                { text: '2', type: 'normal' },
-                { text: '3', type: 'normal' },
-                { text: '+', type: 'orange' },
-                { text: '0', type: 'normal double' },
-                { text: '.', type: 'normal' },
-                { text: '=', type: 'orange' }
-              ]
-    
+
+
+        this.state = {
+            total: null,
+            next: null,
+            operation: null,
+        };
+
+    }
+
+    showContent = (content) => {
+        this.setState(() => ({ next: content }));
+      }
+
+
+    act = (operation) => {
+        console.log(operation)
+        this.setState((state) => (calculate(state, operation)));
       }
 
-      render () {
+    render() {
+        const { next, total, operation } = this.state;
+        
+        const button = [
+            { text: 'AC', type: 'normal' },
+            { text: '+/-', type: 'normal' },
+            { text: '%', type: 'normal' },
+            { text: '÷', type: 'orange' },
+            { text: '7', type: 'normal' },
+            { text: '8', type: 'normal' },
+            { text: '9', type: 'normal' },
+            { text: 'x', type: 'orange' },
+            { text: '4', type: 'normal' },
+            { text: '5', type: 'normal' },
+            { text: '6', type: 'normal' },
+            { text: '-', type: 'orange' },
+            { text: '1', type: 'normal' },
+            { text: '2', type: 'normal' },
+            { text: '3', type: 'normal' },
+            { text: '+', type: 'orange' },
+            { text: '0', type: 'normal double' },
+            { text: '.', type: 'normal' },
+            { text: '=', type: 'orange' }
+        ];
+
         return (
             <div className='cover-area'>
-            <Screen />
-            <div class='calculator-grid'>
-                { this.state.map(btn => (
-               <Button key={btn.text} text={btn.text} type={btn.type} />
-            ) )}
+                <Screen info={next || total || '0'} opr={operation} tx={total} onChange={(e) => this.showContent(e.target.value)} />
+
+                <div class='calculator-grid'>
+                    {button.map(btn => (
+                        <Button key={btn.text} text={btn.text} type={btn.type} act={this.act} />
+                    ))}
+                </div>

```

Comment:
Kindly, make sure to display the operator when doing the calculation. The user have to know what operator they choose. Please implement that.

https://user-images.githubusercontent.com/34578631/177872648-cd8460c0-490f-47aa-92cb-a4f297d47b7f.mov

Reply:
Dear Mihreteab,

Thank you for the feedback. 
Please find below the project requirement. The math operator was not included in the requirement. 

![image](https://user-images.githubusercontent.com/46789910/177944085-459edf4f-59c2-49f3-a5c5-a0c1e7396fa1.png)

Reply:
@ogiogiovictor, On the last bullet point it say test your app in the browser and should work as expected. The operation works fine but the only problem is that the clicked operator is not displayed on the browser as of the numbers. I hope that clarify the point.

Reply:
Dear Mihreteab,

Thank you for the feedback, however, the only said it should be able to perform math operation which the application is already doing.

Reply:
Well resolved 👍🏽